components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,3033 +1,1829 @@
-    use aoa_tracers,        only: aoa_tracers_register
-    use aircraft_emit,      only: aircraft_emit_register
-    use cam_diagnostics,    only: diag_register
-    use cloud_diagnostics,  only: cloud_diagnostics_register
-    use cospsimulator_intr, only: cospsimulator_intr_register
-    use rad_constituents,   only: rad_cnst_get_info ! Added to query if it is a modal aero sim or not
-    use subcol,             only: subcol_register
-    use subcol_utils,       only: is_subcol_on
-    use output_aerocom_aie, only: output_aerocom_aie_register, do_aerocom_ind3
-    use crm_physics,        only: crm_physics_register
-
-    !---------------------------Local variables-----------------------------
-    !
-    integer  :: m        ! loop index
-    integer  :: mm       ! constituent index
-    integer  :: dummy    ! for unused output from pbuf_add_field calls
-    !-----------------------------------------------------------------------
-
-    integer :: nmodes
-
-    logical           :: use_MMF
-    character(len=16) :: MMF_microphysics_scheme
-
-    call phys_getopts(shallow_scheme_out       = shallow_scheme, &
-                      macrop_scheme_out        = macrop_scheme,   &
-                      microp_scheme_out        = microp_scheme,   &
-                      cld_macmic_num_steps_out = cld_macmic_num_steps, &
-                      do_clubb_sgs_out         = do_clubb_sgs,     &
-                      do_aerocom_ind3_out      = do_aerocom_ind3,  &
-                      use_subcol_microp_out    = use_subcol_microp, &
-                      state_debug_checks_out   = state_debug_checks, &
-                      micro_do_icesupersat_out = micro_do_icesupersat, &
-                      pergro_test_active_out   = pergro_test_active, &
-                      pergro_mods_out          = pergro_mods)
-    call phys_getopts( use_MMF_out = use_MMF )
-    call phys_getopts( MMF_microphysics_scheme_out = MMF_microphysics_scheme)
-
-    ! Initialize dyn_time_lvls
-    call pbuf_init_time()
-
-    ! Register the subcol scheme
-    call subcol_register()
-
-    ! Register water vapor.
-    ! ***** N.B. ***** This must be the first call to cnst_add so that
-    !                  water vapor is constituent 1.
-    if (moist_physics) then
-       call cnst_add('Q', mwh2o, cpwv, 1.E-12_r8, mm, &
-            longname='Specific humidity', readiv=.true., is_convtran1=.true.)
-    else
-       call cnst_add('Q', mwh2o, cpwv, 0.0_r8, mm, &
-            longname='Specific humidity', readiv=.false., is_convtran1=.true.)
-    end if
+  !----------------------------------------------------------------------- 
+  ! Purpose: Register constituents and physics buffer fields.
+  !-----------------------------------------------------------------------
+  use physics_buffer,           only: pbuf_init_time
+  use physics_buffer,           only: pbuf_add_field, dtype_r8
+  use shr_kind_mod,             only: r8 => shr_kind_r8
```

Comment:
you already included this in the begging of the module so not necessary to include it here again.

Reply:
good call, will fix.

Reply:
I am all for moving the common use statement up the module level. This is just minor, but I see that some repetitions of the use statements in this file (one inside module, one inside subroutines).

Reply:
I removed all the redundant shr_kind_mod use statements. I can look for more to remove. It doesnt really hurt anything either way, but it's nice to get this as clean and streamlined as possible.